### PR TITLE
#40383 Remove "update benefits" button and make changes automatic

### PR DESCRIPTION
**Pull request by nbarthelemy**
## Description

Remove "update benefits" button and make changes automatic

## Original issue(s)
department-of-veterans-affairs/va.gov-team#40383

## Testing done

- local

## Acceptance criteria
- [ ] Validate changes automatically reflect on the calculator (Your estimated benefits)

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx [hunk 1]
```diff
@@ -97,6 +98,8 @@ function CalculateYourBenefitsForm({
       }
       setInvalidZip('');
       setInputUpdated(true);
+
+      if(!environment.isProduction()) updateEstimatedBenefits();
```
Comment (dsrr47) [inline src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx:102]: looks like a few linting errors to sort out.

Comment (nbarthelemy) [reply src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx]: Fixed

File: src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx [hunk 2]
```diff
@@ -97,6 +98,8 @@ function CalculateYourBenefitsForm({
       }
       setInvalidZip('');
       setInputUpdated(true);
+
+      !environment.isProduction() && updateEstimatedBenefits();
```
Comment (dsrr47) [inline src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx:102]: Linter issue here

File: src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx [hunk 3]
```diff
@@ -107,6 +110,8 @@ function CalculateYourBenefitsForm({
     setInputUpdated(true);
     calculatorInputChange({ field, value });
 
+    !environment.isProduction() && updateEstimatedBenefits();
```
Comment (dsrr47) [inline src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx:113]: Linter Issue here

File: src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx [hunk 4]
```diff
@@ -202,6 +207,7 @@ function CalculateYourBenefitsForm({
     });
     eligibilityChange({ [field]: value });
     setInputUpdated(true);
+    !environment.isProduction() && updateEstimatedBenefits();
```
Comment (dsrr47) [inline src/applications/gi/components/profile/CalculateYourBenefitsForm.jsx:210]: Here

**Review by bah-manish — APPROVED** [review]

**Pull request merged**